symphony/lib/toolkit/class.field.php

```diff
@@ -1538,10 +1538,25 @@ public function buildDSRetrievalSQL($data, &$joins, &$where, $andOperation = fal
     }
 
     /**
+     * Determine if the requested $order is random.
+     *
+     * @since Symphony 2.7.0
+     * @param string $order
+     *  the sorting direction.
+     * @return boolean
+     *  true if the $order is either 'rand' or 'random'
+     */
+    public static function isRandomOrder($order)
```

Comment:
> You should not test protected or private members directly

I must disagree on that! That's the point of unit tests!

> Static methods are nothing more than namespaced global functions.

Agreed. That's how I would classify that function.

> The Field class is not stateless, why should this method be outside them?

The method is. If `Field` had a `$order` property, it would make sense to make it an instance method. And with late static binding, it can even be overridden by child classes.

The only positive thing of making it protected and instance would be that later on, it could be used with DI. I think that's one of the valid argument. 

Please @froschdesign continue to argue with me!!

@brendo @michael-e What are your thoughts on this ? I know we've had the static or not debate often, but since options can change, I am asking for yours once again!